concourse/model/traits/component_descriptor.py

```diff
@@ -383,12 +357,13 @@ def transformer(self):
     def validate(self):
         super().validate()
 
-        ocm_repository_url = self.ctx_repository_base_url()
-        ocm_repository_mappings: list[dict] = self.raw['ocm_repository_mappings']
-        if not ocm_repository_url and ocm_repository_mappings:
-            raise ModelValidationError(
-                'ocm_repository_url must be defined if ocm_repository_mappings are'
-            )
+        if self.ocm_repository:
+            ocm_repository_url = self.ocm_repository.oci_ref
```

Comment:
If `self.ocm_repository` evaluates to `True`, we'll always have an `ocm_repository_url`, don't we?

Reply:
that's correct m(